app/services/SMService.scala

```diff
@@ -171,4 +166,20 @@ trait SMService extends Logging {
   def retrieveServiceLogs(service : String): String = {
     Process(s"sm -l $service").!!.takeRight(100000).trim
   }
+
+  def getConsecutivePorts(searchRange: List[Int], step: Int): List[List[Int]] = {
+    def filterConsecutives(groupedList: List[List[Int]]): List[List[Int]] = {
+      groupedList.filter(subList => subList == (subList.head until subList.head + step).toList)
+    }
+
+    val portsInUse = jsonConnector.loadServicesJson.fields
+      .filter { case (_, js) => js.\("defaultPort").asOpt[Int].isDefined }
+      .map { case (_, js) => js.\("defaultPort").as[Int] }
+
+    val availablePorts = searchRange.filterNot(portsInUse.contains)
+
+    val firstSearch = filterConsecutives(availablePorts.grouped(step).toList)
+
+    if(firstSearch.nonEmpty) firstSearch else filterConsecutives(availablePorts.sliding(step).toList)
```

Comment:
I've refactored the `getConsecutivePorts` method to search based on a grouping of `X` and if that result is empty it will perform a second search and search with sliding as opposed to grouped.